### PR TITLE
Media gallery support

### PR DESCRIPTION
**Pull request by gumaerc**
The static HTML generator does not currently generate HTML for media galleries.  How can we add this functionality?

#### What are the relevant tickets?

https://github.com/mitodl/ocw-data-parser/pull/14 and https://github.com/mitodl/ocw-data-parser/pull/15 should be merged before this, as they contain functionality and fixes key to making the HTML output practically usable at all.

#### What's this PR do?

The primary change with this PR is the addition of handing for media galleries to static_html_generator, however there are a number of other small fixes that were necessary.  A number of additional properties were added to the parser to be pulled in or changed to be named more intelligently, including:

**Pages**
- is_media_gallery

**Embedded Media Pages**
- about_this_resource_text
- related_resources_text
- transcript
- id as "short_url"

**Embedded Media**
- media_asset_type as "type"
- media_location as "media_location" instead of "media_info"

The following functions were added to support generation of media gallery content:

**generate_embedded_media_pages**
This function loops thorough the course embedded media and if it belongs to the page passed into the function, runs generate_embedded_media_page on it.

**generate_embedded_media_page**
Generates the actual content of the embedded media page and writes it to a file.

**generate_embedded_media_playlist**
Generates HTML for a list of thumbnails and links for embedded media pages for a given media gallery.

Other more minor changes include:
- Update CDN links to bootstrap and jquery along with some basic styling to make the pages display somewhat properly
- Fix "resolveuid" links so that they are processed in all cases, including when they don't have "./" prepended to them
- A number of functions including generate_breadcrumbs and generate_side_menu were updated to support a prefix for links so that sub-pages like those in the media gallery can be properly linked to and away from.

#### How should this be manually tested?

Process an OCW course as detailed in the readme that contains a media gallery, such as a "Video Lectures" page.  The resulting HTML should contain a basic representation of the media gallery including embedded and downloadable video and transcripts.

#### Any background context you want to provide?

The actual implementation of the HTML generation code will probably change in the future as we work out more details about the static publishing pipeline.

### CODE REVIEW

File: ocw_data_parser/static_html_generator.py [hunk 1]
```diff
@@ -37,43 +38,51 @@ def generate_index_page(mj, destination):
...
         breadcrumbs = "<li class=\"breadcrumb-item active\"><a href=\"index.html\">%s</a></li>\n" % mj["title"]
-    else:
-        while page_trail[-1]["parent_uid"] != mj["uid"]:
-            for parent_page in mj["course_pages"]:
-                if parent_page["uid"] == page_trail[-1]["parent_uid"]:
-                    page_trail.append(parent_page)
-                    break
-        page_trail.reverse()
-        breadcrumbs = "<li class=\"breadcrumb-item\"><a href=\"index.html\">%s</a></li>\n" % mj["title"]
+    if page_trail:
```
Comment (mbertrand) [inline ocw_data_parser/static_html_generator.py]: `else:`?

File: ocw_data_parser/static_html_generator.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import logging
-
-from .utils import is_json, get_correct_path, load_json_file
+import os
+from .utils import is_json, get_correct_path, get_page_from_url, load_json_file
```
Comment (mbertrand) [inline ocw_data_parser/static_html_generator.py]: `get_page_from_url` doesn't exist in utils

**Review by mbertrand — CHANGES_REQUESTED** [review]

File: ocw_data_parser/static_html_generator.py (hunk 1 above)
Comment (gumaerc) [reply ocw_data_parser/static_html_generator.py]: Right you are.  I forget why I explicitly made a second if statement here, but it should definitely just be an else.

File: ocw_data_parser/static_html_generator.py (hunk 2 above)
Comment (gumaerc) [reply ocw_data_parser/static_html_generator.py]: This is a merge artifact and has been corrected.

File: .vscode/settings.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+    "python.pythonPath": "/usr/bin/python3"
+}
```
Comment (mbertrand) [inline .vscode/settings.json]: Remove from git?

Comment (gumaerc) [reply .vscode/settings.json]: Whoops!

**Review by mbertrand — APPROVED** [review]
👍

**Pull request merged**